### ISSUE TITLE
Version 2025.1.2 does not read the “Formula” tag.

### ISSUE THREAD

**Issue by Neluser** [nvaccess/nvda#18405]
### Actual behavior:

In version 2025.1.2, NVDA does not read the alternative text inside the “Formula” tag. This behavior was observed on three different devices with different documents.

### Expected behavior:

In version 2025.1.2, NVDA can read the alternative text of the Formula tag.

### NVDA logs, crash dumps and other attachments:

[NVDA Logs.txt](https://github.com/user-attachments/files/21032497/NVDA.Logs.txt)

#### NVDA installed/portable/running from source:
NVDA installed
#### NVDA version:
2025.1.2
#### Windows version:
Wjndows 11 Pro 24H2, 26100.4351
#### Name and version of other software in use when reproducing the issue:
Adobe Acrobat (64-bt) (25.001.20531)
#### Does the issue still occur after restarting your computer?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
yes. ver. 2024.4.1. NVDA reads Alternative text of the Formula tag.
#### If NVDA add-ons are disabled, is your problem still occurring?
yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes

**Comment by Neluser** [comment]
Sorry, I meant that it doesn't read the alternative text in the "formula" tag. I need to adjust my report.

**Comment by davidcarlisle** [comment]
Sorry my earlier (deleted) statement was incorrect. The `/Alt` is read if mathcat addon is installed, not otherwise. I don't know if that is the intended behaviour.

A small test pdf 

[alt.pdf](https://github.com/user-attachments/files/21035088/alt.pdf)

**Comment by seanbudd** [comment]
@Neluser can you please provide an example PDF to test with?

**Comment by davidcarlisle** [comment]
with the test file above,   reading via `<nvda>-<down arrow>` nothing gets read after `a formula` if MathCAT is disabled

```
Speaking [CallbackCommand(name=say-all:lineReached), LangChangeCommand ('en_US'), 'testing alt', CallbackCommand(name=say-all:lineReached)]
IO - speech.speech.speak (12:37:25.607) - MainThread (36084):
Speaking [LangChangeCommand ('en_US'), ' me', CallbackCommand(name=say-all:lineReached)]
IO - speech.speech.speak (12:37:25.609) - MainThread (36084):
Speaking [LangChangeCommand ('en_US'), 'July 3, 2025', CallbackCommand(name=say-all:lineReached)]
IO - speech.speech.speak (12:37:25.621) - MainThread (36084):
Speaking [LangChangeCommand ('en_US'), 'a formula', CallbackCommand(name=say-all:lineReached)]
IO - speech.speech.speak (12:37:25.621) - MainThread (36084):
Speaking [CallbackCommand(name=say-all:stop)]
```

but with MathCAT enabled the `/Alt` is passed to MathCAT as `<mtext>` and is read.


```
Speaking [CallbackCommand(name=say-all:lineReached), LangChangeCommand ('en_US'), 'page 1', 'testing alt', CallbackCommand(name=say-all:lineReached)]
IO - speech.speech.speak (12:41:14.811) - MainThread (33508):
Speaking [LangChangeCommand ('en_US'), ' me', CallbackCommand(name=say-all:lineReached)]
IO - speech.speech.speak (12:41:14.814) - MainThread (33508):
Speaking [LangChangeCommand ('en_US'), 'July 3, 2025', CallbackCommand(name=say-all:lineReached)]
DEBUG - NVDAObjects.IAccessible.adobeAcrobat.AcrobatNode._get_mathMl (12:41:14.828) - MainThread (33508):
_get_mathMl (PDF): math recognized: False, child count=0,
  name='LaTeX formula starts \begin {equation*} x^2+y^2 \end {equation*} LaTeX formula ends ', value found from AF ='LaTeX formula starts \begin {equation*} x^2+y^2 \end {equation*} LaTeX formula ends '
DEBUG - NVDAObjects.IAccessible.adobeAcrobat.AcrobatNode._get_mathMl (12:41:14.828) - MainThread (33508):
_get_mathMl: didn't find MathML -- returning value as mtext: <math><mtext>LaTeX formula starts \begin {equation*} x^2+y^2 \end {equation*} LaTeX formula ends </mtext></math>
IO - speech.speech.speak (12:41:14.846) - MainThread (33508):
Speaking [LangChangeCommand ('en_US'), 'a formula', CallbackCommand(name=say-all:lineReached)]
IO - speech.speech.speak (12:41:14.846) - MainThread (33508):
Speaking [LangChangeCommand ('en-us'), ' ', 'LaTeX formula starts \\begin {equation*} x^2+y^2 \\end {equation*} LaTeX formula ends', ' ', '
```
The `value found from AF ` debug message is a bit misleading as it actually took the text from an `/Alt` attribute not an Associated file in this case.

**Comment by Neluser** [comment]
[NVDA test .pdf](https://github.com/user-attachments/files/21099884/NVDA.test.pdf)
Here is the Test Document, in past versions NVDA could read the alternate test from the “Formula” tag, but in the latest version the program is just silent.

**Comment by Neluser** [comment]
> [@Neluser](https://github.com/Neluser) can you please provide an example PDF to test with?

[NVDA test .pdf](https://github.com/user-attachments/files/21100039/NVDA.test.pdf)

Hi, here's an example document

**Comment by SaschaCowley** [comment]
CC @NSoiffer

**Comment by NSoiffer** [comment]
This only happens if you don't have a MathML reader installed. Since MathCAT is going to built into NVDA "soon" ([there's a draft PR](https://github.com/nvaccess/nvda/pull/18323)), it doesn't seem worth writing code looking to see if there is a MathML reader installed. In the meantime, the workaround is to install MathCAT, MathPlayer, or Access8Math. I haven't tested the later two, but they should work.

I suggest the issue be closed.

@Neluser: having alt text for speech is generally a bad idea for accessibility because non-sighted users have no idea what the formula looks like. They will only hear the alt text; the braille will only be the alt text also. $𝐻(𝑃) = 𝑓(𝒙 ) P$ is invisible to them. Using text doesn't result in a very nice display for the sighted user either.

**Comment by Neluser** [comment]
Thank you very much for the information. 
I'm closing the ticket.

**Comment by CyrilleB79** [comment]
I am reopening for tracking for now. The issue is not fixed until MathCAT is actually integrated in NVDA.
This issue will be automatically closed when the corresponding PR will be merged in master.
